cns/restserver/util.go

```diff
@@ -840,3 +840,90 @@ func (service *HTTPRestService) isNCWaitingForUpdate(
+			NetworkContainerID:         ncDetails.CreateNetworkContainerRequest.NetworkContainerid,
+			IPConfiguration:            ncDetails.CreateNetworkContainerRequest.IPConfiguration,
+			Routes:                     ncDetails.CreateNetworkContainerRequest.Routes,
+			CnetAddressSpace:           ncDetails.CreateNetworkContainerRequest.CnetAddressSpace,
+			MultiTenancyInfo:           ncDetails.CreateNetworkContainerRequest.MultiTenancyInfo,
+			PrimaryInterfaceIdentifier: ncDetails.CreateNetworkContainerRequest.PrimaryInterfaceIdentifier,
+			LocalIPConfiguration:       ncDetails.CreateNetworkContainerRequest.LocalIPConfiguration,
+			AllowHostToNCCommunication: ncDetails.CreateNetworkContainerRequest.AllowHostToNCCommunication,
+			AllowNCToHostCommunication: ncDetails.CreateNetworkContainerRequest.AllowNCToHostCommunication,
+		}
+		networkContainers[i] = getNcResp
+		i++
+	}
+	service.RUnlock()
+
+	response := cns.GetAllNetworkContainersResponse{
+		NetworkContainers: networkContainers,
+		Response: cns.Response{
+			ReturnCode: types.Success,
+		},
+	}
+	err := service.Listener.Encode(w, &response)
+	logger.Response(service.Name, response, response.Response.ReturnCode, err)
+}
+
+// handlePostNetworkContainers stores all the NCs (from the request that client sent) into CNS's state file
+func (service *HTTPRestService) handlePostNetworkContainers(w http.ResponseWriter, r *http.Request) {
+	var req cns.PostNetworkContainersRequest
+	err := service.Listener.Decode(w, r, &req)
+	logger.Request(service.Name, &req, err)
+	if err != nil {
+		response := cns.PostNetworkContainersResponse{
+			Response: cns.Response{
+				ReturnCode: types.InvalidRequest,
+				Message:    fmt.Sprintf("[Azure CNS] Create Network Container failed with error: %s", err.Error()),
+			},
+		}
+		err = service.Listener.Encode(w, &response)
+		logger.Response(service.Name, response, response.Response.ReturnCode, err)
+		return
+	}
+
+	returnCode := types.Success
+	returnMessage := ""
+	for i := 0; i < len(req.CreateNetworkContainerRequests); i++ {
+		createNcReq := req.CreateNetworkContainerRequests[i]
+		ncDetails, found := service.getNetworkContainerDetails(createNcReq.NetworkContainerid)
+		// Create NC if it doesn't exist, or it exists and the requested version is different from the saved version
+		if !found || (found && ncDetails.VMVersion != createNcReq.Version) {
+			nc := service.networkContainer
+			if err = nc.Create(createNcReq); err != nil {
+				returnCode = types.UnexpectedError
+				returnMessage = fmt.Sprintf("[Azure CNS] Create Network Container failed with error: %s", err.Error())
+				break
+			}
+		}
+		// Save NC Goal State details
+		saveNcReturnCode, saveNcReturnMessage := service.saveNetworkContainerGoalState(createNcReq)
+
+		// If NC was created successfully, log NC snapshot.
+		if saveNcReturnCode == types.Success {
+			logNCSnapshot(createNcReq)
+		} else {
+			returnCode = saveNcReturnCode
+			returnMessage = saveNcReturnMessage
+			break
+		}
+	}
```

Comment:
setting variables and then breaking is difficult to follow. it's almost akin to a `goto` which should be discouraged. please extract this to a function that returns the response type. Something like:
```go
func (s *HTTPRestService) createOrUpdateNetworkContainers(...) cns.Response
```

such that it can be used in this scope like
```go
resp := createOrUpdateNetworkContainers(req.CreateNetworkContainerRequests)
err = service.Listener.Encode(w, &response)
logger.Response(service.Name, response, response.Response.ReturnCode, err)

```

Reply:
Added a helper function `createNetworkContainers()` that returns the response, making the logic be more clearer.